python/lsst/daf/butler/core/config.py

```diff
@@ -349,32 +406,102 @@ def names(self, topLevelOnly=False):
 
         Returns
         -------
-        names : `list`
-            List of all names present in the `Config`.
-
-        Notes
-        -----
-        This is different than the built-in method `dict.keys`, which will
-        return only the first level keys.
+        names : `list` of `tuple` of `str`
+            List of all names present in the `Config` where each element
+            in the list is a `tuple` of strings representing the hierarchy.
         """
         if topLevelOnly:
-            return list(self.keys())
+            return list((k,) for k in self)
 
-        def getKeys(d, keys, base):
+        def getKeysAsTuples(d, keys, base):
             if isinstance(d, collections.Sequence):
                 theseKeys = range(len(d))
             else:
                 theseKeys = d.keys()
             for key in theseKeys:
                 val = d[key]
-                levelKey = f"{base}.{key}" if base is not None else key
+                levelKey = base + (key,) if base is not None else (key,)
                 keys.append(levelKey)
                 if isinstance(val, (collections.Mapping, collections.Sequence)) and not isinstance(val, str):
-                    getKeys(val, keys, levelKey)
+                    getKeysAsTuples(val, keys, levelKey)
         keys = []
-        getKeys(self.data, keys, None)
+        getKeysAsTuples(self.data, keys, None)
         return keys
 
+    def names(self, topLevelOnly=False, delimiter=None):
+        """Get the delimited name of all the keys in the hierarchy.
```

Comment:
Maybe "a delimited name" instead of "the delimited name" since there isn't one canonical name.